plugins/azure-devops/src/hooks/useUserTeamIds.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { azureDevOpsApiRef } from '../api';
+import { useApi } from '@backstage/core-plugin-api';
+import { useAsync } from 'react-use';
+
+export function useUserTeamIds(userId: string | undefined): {
+  teamIds?: string[];
+  loading: boolean;
+  error?: Error;
+} {
+  const api = useApi(azureDevOpsApiRef);
+
+  const {
+    value: teamIds,
+    loading,
+    error,
+  } = useAsync(() => {
+    return userId ? api.getUserTeamIds(userId) : Promise.resolve(undefined);
```

Comment:
```suggestion
    return userId ? api.getUserTeamIds(userId) : Promise.resolve([]);
```

Reply:
My logic around this is that if there's no `userId` then there's no list of team ids that can be retrieved so `undefined` can represent that in a way. Also I think `useAsync` always returns `T | undefined` so does it make much difference in this case?

Reply:
Ah sure, that's fair.